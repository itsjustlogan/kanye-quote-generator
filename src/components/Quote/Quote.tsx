import React from 'react'

interface Props {
  quote: string
}

export default function Quote({ quote }: Props) {
  return (
    <>
      <blockquote
        data-testid="quote"
        aria-label="quote"
        className="text-center p-8 w-fit text-white bg-gray-900 font-bld mt-36 mx-auto border-radius-medium xl:font-lg"
      >
        &quot;
        {quote}
        &quot;
      </blockquote>
      <p aria-label="quote by" className="text-center p-8 xl:font-lg">
        <i>- Kanye West</i>
      </p>
    </>
  )
}
