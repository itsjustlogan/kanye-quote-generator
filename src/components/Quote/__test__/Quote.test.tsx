import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Quote from '..'

describe('Quote Component', () => {
  afterEach(() => {
    cleanup()
  })

  it('should render correctly', () => {
    expect(Quote).toBeTruthy()

    const { asFragment, rerender } = render(<Quote quote="rendered" />)

    expect(asFragment().textContent).toContain('rendered')

    rerender(<Quote quote="re-rendered" />)
    expect(asFragment().textContent).toContain('re-rendered')
  })

  it('should match screenshot', () => {
    const { asFragment } = render(<Quote quote="very deep quote" />)

    expect(asFragment()).toMatchSnapshot()
  })
})
