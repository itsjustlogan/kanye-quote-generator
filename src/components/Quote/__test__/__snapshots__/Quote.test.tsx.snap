// Vitest Snapshot v1

exports[`Quote Component > should match screenshot 1`] = `
<DocumentFragment>
  <blockquote
    aria-label="quote"
    class="text-center p-8 w-fit text-white bg-gray-900 font-bld mt-36 mx-auto border-radius-medium xl:font-lg"
    data-testid="quote"
  >
    "very deep quote"
  </blockquote>
  <p
    aria-label="quote by"
    class="text-center p-8 xl:font-lg"
  >
    <i>
      - Kanye West
    </i>
  </p>
</DocumentFragment>
`;
